### PR TITLE
Added Flow Section and Initial Architecture Diagram (plus minor README fixes)

### LINKED ISSUE #5: Add an architectural diagram
Internal Requirements Doc:
https://pages.github.ibm.com/IBMCode/IBMCodeContent/patterns/dev/#4-add-an-architectural-diagram

Tasks:
- Write Flow section of README
- Create rough draft of arch diagram

### PR DESCRIPTION
**Pull request by ajbozarth**
- Added Flow Section
- Temporary arch diagram, will be replaced by design team version
- Fixed minor typos found by @kant

Closes #5

### CODE REVIEW

**Comment by ajbozarth** [comment]
@rhagarty and @djalova feed back on both the Flow Steps and the diagram. I'm a bit unsure on both, this is actually the 5th iteration of the flow/arch I've written.

File: README.md [hunk 1]
```diff
@@ -26,11 +26,17 @@ When the reader has completed this Code Pattern, they will understand how to:
...
 ## Flow
 
-<!-- TODO: Add Flow Steps -->
+1. Server initializes by sending default images to Model API and receiving caption data.
+2. Server starts Web UI with default images and caption data.
+3. User interacts with Web UI containing default images and caption data.
+4. User uploads image(s) to the Server via Web UI.
+5. Server sends image(s) to Model API and receives caption data.
+6. Server returns new caption data to Web UI.
+7. Web UI displays the new image(s) and caption data.
```
Comment (rhagarty) [inline README.md]: What you have will work, but I think the diagram is too busy, and you can combine some things into a single step. Maybe have 4 steps, 1 & 4 between server and API, 2 between user and UI, 3 between UI and server. Then for flow, maybe something like:
1) Server requests and receives default images from API
2) User interacts with UI to view default images and request data on uploaded images
3) UI requests image data from server and displays results
4) Server gets caption data from Model API

Comment (ajbozarth) [reply README.md]: I'll try minimizing it a bit more based on this but there's a one incorrect statement in your suggestions though. The defaults images start on the server and it sends them to the model API. I'll take another pass at the flow based on this after I grab lunch.

Comment (rhagarty) [reply README.md]: Sounds good. My flow steps were just to imply relationships. Counting on you to make the correct the content.

Comment (djalova) [reply README.md]: I agree with @rhagarty I think it can be shortened to fewer steps.

**Comment by ajbozarth** [comment]
@rhagarty @djalova pushed an update for you to look at

**Comment by djalova** [comment]
The last part of step 3 happens after step 4. It might be confusing, maybe we can remove it or change the wording to `...update content when data is returned.`
What do you guys think?

**Comment by ajbozarth** [comment]
@djalova that's better, I'll push an update

**Comment by djalova** [comment]
LGTM

**Pull request merged**